### PR TITLE
fix: Fixes #94 by adding missing filter for extrinsics timestamp

### LINKED ISSUE #94: Assertion error in the "Extrinsic tab page" when user selects "timestamp" from the selection box labelled "from extrinsic section"
When select the "Extrinsics tab" and then choose "timestamp" from the selection box under the "from extrinsic section" label it crashes with `Error: Assertion failed`. See screenshots below:

screen shot 2018-06-13 at 10 42 46 am

screen shot 2018-06-13 at 10 43 18 am

### PR DESCRIPTION
**Pull request by ltfschoen**
I have closed the PR that was created from my fork of the repo https://github.com/polkadot-js/apps/pull/154. This new PR is a branch of this repo instead.
Outstanding review comments from @jaco are as follows with progress shown:
* [X] - Additionally, type is not a string, it is one of public or private.
* [X] - Almost there. `any` is a bit of a "smell", only used when no options. Here we can actually type it using Section + generics. (The callers all have the same generic-ified Section type). Solved using jacos tip
* [X] - As above. Additionally, type is not a string, it is one of public or private.

### CODE REVIEW

**Comment by jacogr** [comment]
The label - what are the issues that need fixing?

**Comment by ltfschoen** [comment]
@jacogr I'm having difficulty closing out the review comments that you made by getting rid of the `any` code smell and to use typing with Section + generics instead.

I've pushed a WIP commit with changes so far to https://github.com/polkadot-js/apps/pull/165/commits/038d14facbd23b8f5e0c69b9b249bb790d080ab0, but I'm getting a error `Element implicitly has an 'any' type because type '{}' has no index signature`

File: packages/ui-app/src/util/shouldDisplaySection.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// of the ISC license. See the LICENSE file for details.
+
+import { Section, SectionItems } from '@polkadot/params/types';
+import { SectionVisibilityAll } from '../util/types';
+
+import isUndefined from '@polkadot/util/is/undefined';
+import isNull from '@polkadot/util/is/null';
+
+// methods: <T extends SectionItems<T>>
+function anyMethodToDisplay<SectionItems, K extends keyof SectionItems> (methods: SectionItems[K]) {
```
Comment (jacogr) [inline packages/ui-app/src/util/shouldDisplaySection.ts]:
```
function anyMethodToDisplay <T> (methods: SectionItems<T>) {
```

File: packages/ui-app/src/util/shouldDisplaySection.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      const { isDeprecated, isHidden } = methods[name];
+
+      return !isDeprecated && !isHidden;
+    })
+    .length !== 0;
+}
+
+// section: Section<Storages> or Section<Extrinsics> or Section<Interfaces>
+// type: 'public' | 'private'
+export function shouldDisplaySection<T extends Section<T>, SectionVisibilityAll extends keyof T> (section: T, type: SectionVisibilityAll) {
```
Comment (jacogr) [inline packages/ui-app/src/util/shouldDisplaySection.ts]:
```
export function shouldDisplaySection<T> (section: Section<T>, type: SectionVisibilityAll) {
```

**Comment by jacogr** [comment]
@ltfschoen Check the latest comments.

**Comment by ltfschoen** [comment]
@jacogr ready for re-review/approval now that PR incorporates your feedback

**Pull request merged**